src/pipx/commands/run.py

```diff
@@ -189,14 +188,16 @@ def run(
         # we can't parse this as a package
         package_name = app
 
-    if spec is not None:
-        content = None
-    else:
-        content = maybe_script_content(app, is_path)
+    if python and not Path(python).is_file():
+        py_launcher = find_py_launcher_python(python)
+        if py_launcher:
+            python = py_launcher
```

Comment:
```suggestion
        py_launcher_python = find_py_launcher_python(python)
        if py_launcher_python:
            python = py_launcher_python
```

For clarity, otherwise this reads like we’re assigning _the launcher_ to `python`.

Reply:
Thanks, I agree that is much more clear.